src/Contract/Operation/Matchable.php

```diff
@@ -20,9 +19,7 @@ interface Matchable
 {
     /**
      * @param callable(T, TKey, Iterator<TKey, T>): bool $callback
-     * @param null|callable(T, TKey, Iterator<TKey, T>): T $matcher
-     *
-     * @return Collection<int, bool>
+     * @param null|callable(T, TKey, Iterator<TKey, T>): bool $matcher
```

Comment:
this is now correct I believe? both the callback and the matcher should return booleans

Reply:
Yes that's how it supposed to work.